client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -376,6 +408,11 @@ private StreamCut computeEndStreamCut(Stream stream, Map<Segment, Long> endSegme
                       .thenApply(v -> completeCheckpointAndFetchStreamCut(checkpointId));
     }
 
+    @Override
+    public void updateTruncationStreamCut(Stream stream, StreamCut streamCut) {
+        controller.updateSubscriberStreamCut(stream.getScope(), stream.getStreamName(), groupName, streamCut);
```

Comment:
if this api returns a CompletableFuture then this change needs to be handled differently. The user will be under the impression that this is a synchronous method.

Reply:
The api would not return a CompletableFuture. Wrapped the controller call in `getAndHandleExceptions` to handle the it better.